pkg/lib/botprotection/provider.go

```diff
@@ -44,17 +45,14 @@ func (p *Provider) verifyTokenByCloudflare(token string) error {
 	if p.CloudflareClient == nil {
 		return fmt.Errorf("missing cloudflare credential")
 	}
-	result, err := p.CloudflareClient.Verify(token, string(p.RemoteIP))
+	successResp, err := p.CloudflareClient.Verify(token, string(p.RemoteIP))
 	if err != nil {
+		p.Logger.WithField("cloudflare verification error:", err)
 		return err
 	}
-	if !*result.Success {
-		p.Logger.WithField("cloudflare verification error-codes:", result.ErrorCodes)
-		isServiceUnavailable := p.CloudflareClient.IsServiceUnavailable(result)
-		if isServiceUnavailable {
-			return ErrVerificationServiceUnavailable
-		}
-		return ErrVerificationFailed
+	if successResp == nil {
+		err = fmt.Errorf("cloudflare no error but no success response")
+		return errors.Join(err, ErrVerificationFailed)
```

Comment:
No need this checking. If err is nil, then resp must be present. It is a programming error if this does not hold. Programming error is supposed to be caught by unit tests. We do not handle programming error this way. And of course, we do not report programming error as ErrVerificationFailed. They are entirely different.